src/mixins/socket.js

```diff
@@ -374,7 +375,7 @@ export default {
                     this.storage().token = res.token;
                     this.socket.token = res.token;
                     this.loggedIn = true;
-                    this.username = this.getJWTPayload()?.username;
+                    Object.assign(this, this.getJWTPayload());
```

Comment:
Assigning variables this way sounds like a pretty simple footgun to me.
Let's say further down the line we add a field to the JWTPayload ⇒ This would be added here (potentially overriding stuff) as well without being obvious to other devs.
Could you extract the fields you need instead? ^^

Reply:
It will become a bit too verbose when adding the whole user in the token... but fine.